packages/server-wallet/src/wallet/index.ts

```diff
@@ -61,6 +62,10 @@ type Message = SingleChannelOutput | MultipleChannelOutput;
 const isSingleChannelMessage = (message: Message): message is SingleChannelOutput =>
   'channelResult' in message;
 
+export interface NotificationReceiver {
+  onWalletNotification(message: SingleChannelOutput): void;
+}
```

Comment:
This looks suspiciously like an event emitter. I'm wondering if it would be worth using an event emitter (e.g eventemitter3, like we do in the channel provider and xstate wallet). We could then expose `onNotification` (or event `onFundingUpdated`) on the wallet api, and any receivers could subscribe to it themselves. It seems like that would give us a load of goodies that we might need in the future (unsubscribe, multiple-subscribers etc.) out-of-the-box.

Reply:
I like the event emitter idea. Check out 2826a7021